examples/nifgen/basic-arbitrary-waveform.py

```diff
@@ -0,0 +1,171 @@
+        any_error = True
+        ThrowOnError (vi, status)
+
+# Converts an error code returned by NI-FGEN into a user-readable string.
+def ThrowOnError (vi, error_code):
+    error_message_request = nifgen_types.ErrorMessageRequest(
+        vi = vi,
+        error_code = error_code
+    )
+    error_message_response = nifgen_service.ErrorMessage(error_message_request)
+    raise Exception (error_message_response.error_message)
+
+try:
+    # Initalize NI-FGEN session
+    init_with_channels_resp = nifgen_service.InitWithOptions(nifgen_types.InitWithOptionsRequest(
+        session_name = session_name,
+        resource_name = resource,
+        option_string = options
+    ))
+    vi = init_with_channels_resp.vi
+    CheckForError(vi, init_with_channels_resp.status)
+
+    # Configure channels
+    config_channels_resp = nifgen_service.ConfigureChannels(nifgen_types.ConfigureChannelsRequest(
+        vi = vi,
+        channels = channel_name
+    ))
+    CheckForError(vi, config_channels_resp.status)
+
+    # Configure output mode
+    config_out_resp = nifgen_service.ConfigureOutputMode(nifgen_types.ConfigureOutputModeRequest(
+        vi = vi,
+        output_mode = 1
+    ))
+    CheckForError(vi, config_out_resp.status)
+
+    # Create waveform
+    create_waveform_resp = nifgen_service.CreateWaveformF64(nifgen_types.CreateWaveformF64Request(
+        vi = vi,
+        channel_name = channel_name,
+        waveform_data_array = sine
+    ))
+    waveform_handle = create_waveform_resp.waveform_handle
+    CheckForError(vi, create_waveform_resp.status)
+
+    # Configure arbitrary waveform
+    config_waveform_resp = nifgen_service.ConfigureArbWaveform(nifgen_types.ConfigureArbWaveformRequest(
+        vi = vi,
+        channel_name = channel_name,
+        waveform_handle = waveform_handle,
+        gain = gain,
+        offset = dc_offset
+    ))
+    CheckForError(vi, config_waveform_resp.status)
+
+    # Configure clockmode to VAL_HIGH_RESOLUTION
+    config_clckmode_resp = nifgen_service.ConfigureClockMode(nifgen_types.ConfigureClockModeRequest(
+        vi = vi,
+        clock_mode = 0
+    ))
+    CheckForError(vi, config_clckmode_resp.status)
+
+    # Configure sample rate
+    config_sample_rate_resp = nifgen_service.ConfigureSampleRate(nifgen_types.ConfigureSampleRateRequest(
+        vi = vi,
+        sample_rate = sample_rate
+    ))
+    CheckForError(vi, config_sample_rate_resp.status)
+
+    # Configure output enabled
+    config_outenbl_resp = nifgen_service.ConfigureOutputEnabled(nifgen_types.ConfigureOutputEnabledRequest(
+        vi = vi,
+        channel_name = channel_name,
+        enabled = True
+    ))
+    CheckForError(vi, config_outenbl_resp.status)
+
+    # Initiate generation
+    init_gen_resp = nifgen_service.InitiateGeneration(nifgen_types.InitiateGenerationRequest(
+        vi = vi
+    ))
+    CheckForError(vi, init_gen_resp.status)
+
+    print(f"Generating sine wave at {sample_rate} Hz")
```

Comment:
So this example configures FGEN to generate sine wave and quit? That seems little bit... incomplete. 

@harsha-bhushan can we add some APIs to read output from FGEN and plot the graph so that users can visualize the signal they are generating and also so that they can tweak few configuration parameters in this example and visualize the effect? If that's something that is not possible (probably because it needs something to be wired to fgen output?), is there other way users can visualize what they have configured fgen for, perhaps by using some SFP (in MAX, Instrument Studio)? We can include instructions to visualize the generated waveform in `print` statement here as well as in the documentation above.

Reply:
yes, it wont be possible to measure the output of the device & plot it with just Fgen alone. However, we could plot the waveform_data_array. That is what is done in Labview example too. This will not be a running waveform, but a static waveform of fixed size. 
Additionally you can also include 'Close the window to stop generation' in your print statement. & remove taking keyboard input in the next line.

Reply:
I added the plotting of the wave

Reply:
@harsha-bhushan closing the plot doesn't stop generation though. Did you also meant to call APIs to stop signal generation on FGEN on close of the plot? Also should we change title of the graph to "Sample sine wave" (or something along those lines) to make it clear that we are not showing real waveform from the hardware? @ShantanuShinde what title and axis labels are used in LabVIEW example?

Reply:
Yes, I meant calling necessary APIs after closing the graph to stop generation. 
I like your suggestion of having 'Sample' in the title. I would suggest a more generic "Sample waveform" so that it is inline with example title & represents the capability of the example.
Unfortunately Labview examples dont have axis labels. The current labels do accurately represent the data being plotted, so I'm OK retaining these labels.

Reply:
changed the title to 'Sample Waveform'